src/main/java/cohen/worldcitysearch/WorldCitiesList.java

```diff
@@ -0,0 +1,78 @@
+package cohen.worldcitysearch;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+
+public class WorldCitiesList
+{
+    private final ArrayList<WorldCity> worldCities;
+
+    public WorldCitiesList() throws IOException
+    {
+        this.worldCities = new ArrayList<>();
+
+        File csvData = new File("src/main/resources/worldcities.csv");
+        CSVParser parser = CSVParser.parse(csvData, Charset.defaultCharset(), CSVFormat.RFC4180);
+
+        List<CSVRecord> cities = parser.getRecords();
+        int totalRecords = cities.size();
+
+        for (int i = 1; i < totalRecords; i++)
+        {
+            String city = cities.get(i).get(1);
+            String lat = cities.get(i).get(2);
+            String lon = cities.get(i).get(3);
+            WorldCity worldCity = new WorldCity(
+                    city,
+                    Double.parseDouble(lat),
+                    Double.parseDouble(lon)
+            );
+            worldCities.add(worldCity);
+        }
+    }
+
+    public ArrayList<WorldCity> getList()
+    {
+        return worldCities;
+    }
+
+    public WorldCity getClosestCity(double lat, double lon)
+    {
+        //what about kd-trees?? bounding boxes? to make this faster. Although seems like overall
+        //no way to get past O(n) time in worst case scenario
+
+        WorldCity requestCity = new WorldCity("", lat, lon);
+
+        WorldCity closest = worldCities.get(0);
+
+        for (int i = 1; i < worldCities.size(); i++)
+        {
+            double currentDist = distance(requestCity, closest);
+            WorldCity currentCity = worldCities.get(i);
+
+            double newDistance = distance(requestCity, currentCity);
+
+            if (newDistance < currentDist && newDistance != 0)
```

Comment:
why `newDistance != 0`?

Reply:
The list may include the requested city, and a distance of 0 would be less than anything else, which would lead to the conclusion that the city itself is closest to itself...

Reply:
Yes, you could be in the city which would make it the closest city...